### PR TITLE
MGDOBR-1034: UI E2E test clean up hooks refactoring

### PR DESCRIPTION
**Pull request by sjamboro**
https://issues.redhat.com/browse/MGDOBR-1034

The script probably does not work correctly because only the version on the main branch is executed.

### CODE REVIEW

**Comment by sjamboro** [comment]
Hi @Srihari1192, I will test this approach on Thursday locally, because as you can see the changes in this workflow are not allowed to perform under PR. Please let me know if I missed something or if you prefer a different approach.

We discussed a creation of the list of all bridges and rewrite the bash script to Cypress test. Such approach could be tested by PRs but its definition can took much more time than this approach.

File: .github/workflows/quality-checks-e2e.yml [hunk 1]
```diff
@@ -49,6 +50,8 @@ jobs:
           CYPRESS_PASSWORD: ${{ secrets.CY_PASSWORD }}
           CYPRESS_SANDBOX_DEV_REST_URL: ${{ secrets.CY_SANDBOX_DEV_REST_URL }}
           CYPRESS_SANDBOX_DEV_REST_PATH: ${{ secrets.CY_SANDBOX_DEV_REST_PATH  }}
+      - name: Clear data
```
Comment (sjamboro) [inline .github/workflows/quality-checks-e2e.yml:53]: Add `if: allways()` ... we have to run this script when `E2E tests` step failed

**Comment by Srihari1192** [comment]
@sjamboro Thanks for the PR..Sure will take a look

File: scripts/delete-bridges.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    echo "No processors in Bridge $i."
+    BRIDGE_STATUS=$(curl -s -H "Authorization: $OB_TOKEN" -X GET $MANAGER_URL/api/smartevents_mgmt/v1/bridges/$i | jq '.status' | tr -d \")
+    if [ "$BRIDGE_STATUS" == "ready" ] || [ "$BRIDGE_STATUS" == "failed" ]; then
+      echo "Deleting Bridge $i"
+      curl -s -H "Authorization: $OB_TOKEN" -X DELETE $MANAGER_URL/api/smartevents_mgmt/v1/bridges/$i
+    else
+      echo "Bridge $i status $BRIDGE_STATUS is incompatible with automated deletion, please remove it manually"
+    fi
+  else
+    echo "$NUMBER_OF_PROCESSORS processors found in Bridge $i"
```
Comment (Srihari1192) [inline scripts/delete-bridges.sh:24]: So far In our E2E tests we are not creating any processors, but in future we may add. 
In that case , we need to rework this script to delete bridge after processor deletion too.

Comment (sjamboro) [reply scripts/delete-bridges.sh:24]: @Srihari1192 I will take a look at it, yesterday I have worked on the concept functionality. It seems that hash generated by CI is visible from Cypress tests and even if E2E failed than the clean hook is performed correctly. Delete all instances which contains "testui-$CYPRESS_SUITE_HASH".

File: scripts/delete-bridges.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# Script used to delete Bridges and Processors used in E2E tests
+# Script will delete Bridges without Processors, for Bridges with Processors it will delete Processors if they are in proper state
+# Test Bridges are identified using "test-" prefix in their names
+
+# If Bridge contains Processors then only Processors are deleted by the script execution. If user wants to delete also Bridges then the script has to be executed again after some time (so the Processors can be removed first)!
+
+BRIDGES=($(curl -H "Authorization: $OB_TOKEN" -X GET $MANAGER_URL/api/smartevents_mgmt/v1/bridges | jq '.items[] | select(.name | startswith('\"testui-1f65f\"')) | .id' | tr -d \"))
```
Comment (sutaakar) [inline scripts/delete-bridges.sh:9]: `testui-1f65f` seems hardcoded, shouldn't be there a variable?

Comment (sjamboro) [reply scripts/delete-bridges.sh]: @sutaakar do you think that is better to create new script with the for cycle and in CI leave only call of one script?

Comment (sutaakar) [reply scripts/delete-bridges.sh]: yes, it may be better
also it may be worth splitting the script into dedicated bash methods to increase readability

Comment (sjamboro) [reply scripts/delete-bridges.sh]: I thought about the splitting the script to methods. Currently I am not familiar enough with bash methods (variables visibilities, good practice, etc.) to split it.

**Comment by sjamboro** [comment]
@sutaakar @Srihari1192 @vpellegrino can you review this PR?

`Dev :: Quality Checks` does not work properly right now (migration to v2 has been started)

**Review by sutaakar — APPROVED** [review]

File: scripts/delete-bridges.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# Script used to delete Bridges and Processors used in E2E tests
+# Script will delete Bridges without Processors, for Bridges with Processors it will delete Processors if they are in proper state
+# Test Bridges are identified using "test-" prefix in their names
+
+# If Bridge contains Processors then only Processors are deleted by the script execution. If user wants to delete also Bridges then the script has to be executed again after some time (so the Processors can be removed first)!
+
+BRIDGES=($(curl -s -H "Authorization: $OB_TOKEN" -X GET $MANAGER_URL/api/smartevents_mgmt/v1/bridges | jq '.items[] | select(.name | startswith('\"testui-$CYPRESS_SUITE_HASH\"')) | .id' | tr -d \"))
```
[the remark this answers is not in the document]
Comment (sjamboro) [reply scripts/delete-bridges.sh:9]: @Srihari1192 v2 does not work properly right now. I agree with you that E2E test suites and this script should work with the same version of API. CI contains this SANDBOX_DEV_REST_PATH variable. I suggest to:
- concatenate DEV_REST_URL and SEV_REST_PATH in this PR
- I will create JIRA to update this SANDBOX_DEV_REST_PATH when v2 will work and will be covered by IT in the sandox repository. 
- I can add some verification that this script will fail when on of the important environment variable is missing.
What do you think?

There is example of execution when all E2E tests are broken by v2 and the hook is still working correctly
image

[the remark this answers is not in the document]
Comment (Srihari1192) [reply scripts/delete-bridges.sh:9]: @sjamboro Sure create a Jira for that and will update once V2 works

**Review by Srihari1192 — APPROVED** [review]

**Comment by sjamboro** [comment]
I have set label to `do not merge` because from some reason the output of the script is: ```Run OB_TOKEN="***" MANAGER_URL=*** PATH=*** ./scripts/delete-bridges-in-loop.sh
Run script #1:
Info: The token is set correctly
Info: The hash is set correctly
Info: The manager url is set correctly
Info: The path is set correctly
./scripts/delete-bridges.sh: line [13](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/actions/runs/3659416205/jobs/6185394061#step:11:14): curl: command not found
./scripts/delete-bridges.sh: line 13: jq: command not found
./scripts/delete-bridges.sh: line 13: tr: command not found
Script found 0 bridge(s) for hash e44[15](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/actions/runs/3659416205/jobs/6185394061#step:11:16) on url ***
Clean rutine exits after 1 performation(s)```

**Comment by sutaakar** [comment]
@sjamboro Thinking whether the issue is caused by quotes....
Can you try:
`BRIDGES=($(curl -s -H "Authorization: $OB_TOKEN" -X GET $MANAGER_URL$PATH/bridges | jq -r '.items[] | select(.name | startswith("testui-$CYPRESS_SUITE_HASH")) | .id' ))`

**Comment by sjamboro** [comment]
> @sjamboro Thinking whether the issue is caused by quotes.... Can you try: `BRIDGES=($(curl -s -H "Authorization: $OB_TOKEN" -X GET $MANAGER_URL$PATH/bridges | jq -r '.items[] | select(.name | startswith("testui-$CYPRESS_SUITE_HASH")) | .id' ))`
@sutaakar I have tried `curl -H "Authorization: $OB_TOKEN" -X GET $CYPRESS_SANDBOX_DEV_REST_URL$CYPRESS_SANDBOX_DEV_REST_PATH/bridges | jq '.items[] | select(.name | startswith('\"testui-$CYPRESS_SUITE_HASH\"')) | .id' | tr -d \"` locally. It returned the empty list but it does not failed with ```./scripts/delete-bridges.sh: line 13: curl: command not found
./scripts/delete-bridges.sh: line 13: jq: command not found
./scripts/delete-bridges.sh: line 13: tr: command not found```

**Comment by vpellegrino** [comment]
Waiting for the CI to be green before reviewing

**Pull request closed without merging**